packages/studio-be/src/core/users/workspace-service.ts

```diff
@@ -84,6 +84,24 @@ export class WorkspaceService {
     return list.find(x => x.workspace === workspace)
   }
 
+  async save(workspaces: Workspace[]): Promise<void> {
+    return this.ghost.global().upsertFile('/', 'workspaces.json', JSON.stringify(workspaces, undefined, 2))
+  }
+
+  async addBotRef(botId: string, workspaceId: string): Promise<void> {
+    const workspaces = await this.getWorkspaces()
+    const workspace = workspaces.find(x => x.id === workspaceId)
+
+    if (!workspace) {
+      throw new Error(`Specified workspace "${workspaceId}" doesn't exist`)
+    }
+
+    if (!workspace.bots.includes(botId)) {
+      workspace.bots.push(botId)
+    }
+    return this.save(workspaces)
```

Comment:
You should only save the workspace if it was updated otherwise it calls the ghost for nothing.